### ISSUE TITLE
Require environment variables for TFE provider

### ISSUE THREAD

**Issue by hcrhall** [terraform-providers/terraform-provider-tfe#31]
It would be good to include an environment variable for the TFE provider so that it is consistent with all of the other providers that are currently available. Something like TFE_HOSTNAME and TFE_TOKEN would suffice.

**Comment by svanharmelen** [comment]
@hcrhall we did discuss this when designing the provider, but decided not to support environment variables but instead rely on the [CLI Config File](https://www.terraform.io/docs/commands/cli-config.html#credentials).

This file is meant as a central location where you can manage all TFE related credentials. It was already used for module registry credentials and so it made a lot of sense to use the same approach for the TFE provider as well. Especially knowing that in most cases the creds used for the registry and TFE are the same creds.

But, I must admit that personally I do see value in supporting environment variables as well so I'll have another look at this and discuss it internally as well. So to be continued.

**Comment by hcrhall** [comment]
Thanks @svanharmelen. My use case is for integrating with CICD so my preference is to have everything set as a secure variable. The CLI config file works great if you are working as a single practitioner on your workstation but doesn't translate to CICD because I have to have a step that writes the file to my build agent. If we could just have environment variables for setting tokens throughout (remote operations backend, TFE provider etc.) that would be amazing.

**Comment by robinbowes** [comment]
I'd like to see this too.

**Comment by djaboxx** [comment]
ATLAS_TOKEN seems to work in place of TFE_TOKEN. I'm using it here, https://github.com/HappyPathway/AWSSentinel (notice the provider config is empty). The hostname parameter though is a different story, as it defaults to TFE SaaS. Terraform does allow setting environment variables; but, the name would have to be pre-pended with TF_VAR. So, for instance, the tfe_hostname variable for the below configuration would need to be set as TF_VAR_tfe_hostname. 

```hcl
# Configure the Terraform Enterprise Provider
provider "tfe" {
  hostname = "${var.tfe_hostname}"
}
```

Hope this helps.

**Comment by tdi** [comment]
+1 from me too

**Comment by alisdair** [comment]
Fixed in #78 and released just now in v0.9.1. The two variables are `TFE_HOSTNAME` and `TFE_TOKEN`, [described in the Authentication section of the docs](https://github.com/terraform-providers/terraform-provider-tfe/blob/master/website/docs/index.html.markdown#authentication).
